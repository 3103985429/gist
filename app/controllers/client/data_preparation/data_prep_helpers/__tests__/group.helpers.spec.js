import {
    getGroupByDurationParams,
    getGroupByRangeParams,
    getGroupByColumnParams,
} from '../group.helpers';
import {
    DATE_DURATION_DECADE,
    DATE_DURATION_YEAR,
    DATE_DURATION_MONTH,
    DATE_DURATION_WEEK,
    DATE_DURATION_DAY,
} from '../../../../../config/dataTypes/date.config';


describe('group helpers', () => {
    describe('getGroupByDurationParams', () => {
        it('should return decade params', () => {
            const result = getGroupByDurationParams(DATE_DURATION_DECADE, 'order');
            expect(result).toMatchSnapshot();
        });

        it('should return month params', () => {
            const result = getGroupByDurationParams(DATE_DURATION_MONTH, 'order');
            expect(result).toMatchSnapshot();
        });

        it('should return week params', () => {
            const result = getGroupByDurationParams(DATE_DURATION_WEEK, 'order');
            expect(result).toMatchSnapshot();
        });

        it('should return day params', () => {
            const result = getGroupByDurationParams(DATE_DURATION_DAY, 'order');
            expect(result).toMatchSnapshot();
        });

        it('should return year params', () => {
            const result = getGroupByDurationParams(DATE_DURATION_YEAR, 'order');
            expect(result).toMatchSnapshot();
        });

        it('should return default params', () => {
            const result = getGroupByDurationParams('', 'order');
            expect(result).toMatchSnapshot();
        });
    });

    describe('getGroupByRangeParams', () => {
        it('should return range params', () => {
            const result = getGroupByRangeParams('100s', 'size');
            expect(result).toMatchSnapshot();
        });
    });

    describe('getGroupByColumnParams', () => {
        it('should return column params', () => {
            const result = getGroupByColumnParams('name');
            expect(result).toMatchSnapshot();
        });
    });
});
