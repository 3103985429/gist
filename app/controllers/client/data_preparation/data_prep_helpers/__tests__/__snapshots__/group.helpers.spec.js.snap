// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group helpers getGroupByColumnParams should return column params 1`] = `
Object {
  "groupIdQueryHelper": "$rowParams.name",
}
`;

exports[`group helpers getGroupByDurationParams should return day params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.Day",
  },
}
`;

exports[`group helpers getGroupByDurationParams should return decade params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.Decade",
  },
}
`;

exports[`group helpers getGroupByDurationParams should return default params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.",
  },
}
`;

exports[`group helpers getGroupByDurationParams should return month params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.Month",
  },
}
`;

exports[`group helpers getGroupByDurationParams should return week params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.Week",
  },
}
`;

exports[`group helpers getGroupByDurationParams should return year params 1`] = `
Object {
  "filterDateFormat": "YYYY",
  "groupByDateFormat": "Y",
  "groupByUnit": "years",
  "groupIdQueryHelper": Object {
    "$year": "$rowParams.Year",
  },
}
`;

exports[`group helpers getGroupByRangeParams should return range params 1`] = `
Object {
  "groupIdProject": Object {
    "$multiply": Array [
      Object {
        "$floor": "$_id",
      },
      1000,
    ],
  },
  "groupIdQueryHelper": Object {
    "$floor": Object {
      "$divide": Array [
        "$rowParams.100s",
        1000,
      ],
    },
  },
  "groupNumber": 1000,
}
`;
